Ghidra/Features/GnuDemangler/src/main/java/ghidra/app/util/demangler/gnu/GnuDemanglerOptions.java

```diff
@@ -43,61 +43,133 @@
 	 */
 	public static final String GNU_DEMANGLER_DEFAULT = GNU_DEMANGLER_V2_33_1;
 
-	private String demanglerName = GNU_DEMANGLER_DEFAULT;
-	private String demanglerApplicationArguments;
+	private final GnuDemanglerFormat format;
+	private final boolean deprecated;
 
 	public GnuDemanglerOptions() {
 		// use default values
+		this(GnuDemanglerFormat.AUTO);
+	}
+
+	public GnuDemanglerOptions(GnuDemanglerFormat format) {
+		this.format = format;
+		// default to the "new" demangler if the format is available in both
+		this.deprecated = format.isDeprecated();
+	}
+
+	public GnuDemanglerOptions(GnuDemanglerFormat format, boolean deprecated) {
+		this.format = format;
+		this.deprecated = deprecated;
+		if (!isValidateFormat(format, deprecated)) {
+			throw new IllegalArgumentException(
+				format.name() + " is not available in the "+getDemanglerName());
+		}
 	}
 
 	public GnuDemanglerOptions(DemanglerOptions copy) {
 		super(copy);
 
 		if (copy instanceof GnuDemanglerOptions) {
 			GnuDemanglerOptions gCopy = (GnuDemanglerOptions) copy;
-			demanglerName = gCopy.demanglerName;
-			demanglerApplicationArguments = gCopy.demanglerApplicationArguments;
+			format = gCopy.format;
+			deprecated = gCopy.deprecated;
+		} else {
+			format = GnuDemanglerFormat.AUTO;
+			deprecated = false;
 		}
 	}
 
+	private GnuDemanglerOptions(GnuDemanglerOptions copy, GnuDemanglerFormat format,
+			boolean deprecated) {
+		super(copy);
+		this.format = format;
+		this.deprecated = deprecated;
+	}
+
 	/**
 	 * Returns the external demangler executable name to be used for demangling.  The 
 	 * default value is {@link #GNU_DEMANGLER_DEFAULT}.
 	 * @return the name
 	 */
 	public String getDemanglerName() {
-		return demanglerName;
+		return deprecated ? GNU_DEMANGLER_V2_24 : GNU_DEMANGLER_V2_33_1;
 	}
 
 	/**
 	 * Sets the external demangler executable name to be used for demangling
 	 * @param name the name
 	 */
```

Comment:
We also add the @Deprecated javadoc tag, along with a note to indicate what to use instead.

Of course, as per my note below, if you remove the method, this does not apply.

Reply:
This one is actually needed as it is used to get the correct executable to use.